notification-service/src/main/java/org/java/mentorship/notification/exception/ValidatorHandler.java

```diff
@@ -0,0 +1,23 @@
+package org.java.mentorship.notification.exception;
+
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.java.mentorship.notification.exception.domain.NotificationServiceError;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnCheckpointRestore;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.java.mentorship.notification.exception.domain.NotificationServiceError;
+
+@RestControllerAdvice
+public class ValidatorHandler {
+    @ExceptionHandler
+    public ResponseEntity<NotificationServiceError> handleException(final MethodArgumentNotValidException ex, final HttpServletRequest request) {
+        return ResponseEntity.badRequest().body(
+                new NotificationServiceError(ex.getBindingResult().getFieldErrors().getFirst().getDefaultMessage())
```

Comment:
What if we've got multiple errors?
I think we could display all of them.
Thus the Exception can have a List<String> errors

Reply:
I think the error message should be kept as a single `String`, the error contract from #60 only has one `error: String` field. I would recommend that the errors' messages are joined into a single string separated by `;`.